src/routes/user.route.tsx

```diff
@@ -0,0 +1,132 @@
+import { signupUserThunk, updateUserThunk, verifyUserThunk } from '@/redux/cognito/cognito.thunk';
+import { RootState } from '@/redux/store';
+import Button from '@vini-vici/viddi/dist/button/button.component';
+import FormField from '@vini-vici/viddi/dist/formfield/formfield.component';
+import Input from '@vini-vici/viddi/dist/input/input.component';
+import React, { useEffect } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { Redirect } from 'react-router-dom';
+
+export default function UserRoute(): React.ReactElement{
+
+    const {
+        loggedInUser,
+        needsVerification
+      } = useSelector(({ cognito: { username, needsVerification, idToken  } }: RootState) => ({ loggedInUser: username, needsVerification, idToken }));
+    
+      const dispatch = useDispatch();
+
+      const [updateData, setUpdateData] = React.useState({
+        password: '',
+        confirmPassword: '',
+        email: '',
+        preferedUserName: ''
+      });
+
+    const [token, setToken] = React.useState('');
+
+    const [validations, setValidations] = React.useState({
+        password: true
+    });
+
+    const [redirect, setRedirect] = React.useState(false);
+    const [error, setError] = React.useState('');
+
+    useEffect(() => {
+        setValidations({
+          password: (updateData.password.length > 0 && updateData.confirmPassword.length > 0) && updateData.password === updateData.confirmPassword
```

Comment:
In testing this looks like in order to update the user's preferred name that I would have to update their password as well.